File: java/0001_Two_Sum.java
import java.util.stream.IntStream;
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int[] res = new int[2];
        int[] temp = new int[nums.length];

        System.arraycopy(nums, 0, temp, 0, nums.length);
        Arrays.sort(nums);
        for (int i = 0, j = nums.length - 1; i < j;) {
            if (nums[i] + nums[j] < target) {
                i++;
            } else if (nums[i] + nums[j] > target) {
                j--;
            } else {
                final int num1 = nums[i];
                final int num2 = nums[j];
                int index1 = IntStream.range(0, temp.length).filter(x -> temp[x] == num1).findFirst().orElse(-1);
                int index2 = IntStream.range(0, temp.length).filter(x -> temp[x] == num2 && x != index1).findFirst().orElse(-1);
                res[0] = index1;
                res[1] = index2;
                return res;
            }
        }
        return null;
    }
    
}

File: python/0001_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # list = []
        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i]+nums[j] == target:

        #             return [i, j]
        mapping = {}
        for i in range(len(nums)):
            if target - nums[i] in mapping:
                return [mapping[target - nums[i]], i]
            mapping[nums[i]] = i